data_model/src/events/data/filters.rs

```diff
@@ -94,50 +96,81 @@ impl Filter for EntityFilter {
     }
 }
 
-#[derive(
-    Clone,
-    PartialOrd,
-    Ord,
-    PartialEq,
-    Eq,
-    Debug,
-    Decode,
-    Encode,
-    Deserialize,
-    Serialize,
-    IntoSchema,
-    Hash,
-)]
-/// Filter for identifiers
-///
-/// Passes id trough filter, if it equals to the one provided in construction
-pub struct IdFilter<Id: Eq>(Id);
-
-impl<Id: Eq> IdFilter<Id> {
-    /// Construct new `IdFilter`
-    pub fn new(id: Id) -> Self {
-        Self(id)
+#[derive(Clone, PartialOrd, Ord, Eq, Debug, Decode, Encode, Serialize, IntoSchema)]
+/// Filter that accepts a data event with the matching origin.
+pub struct OriginFilter<T: HasOrigin>(<T::Origin as Identifiable>::Id)
+where
+    <T::Origin as Identifiable>::Id:
+        Debug + Clone + Eq + Ord + Hash + Decode + Encode + Serialize + IntoSchema;
```

Comment:
now that you've decided in favor of this, it's likely you don't need all these bounds on a struct definition?